### PR TITLE
use nodeport instead of ingress address to access rancher at runtime.  

### PR DESCRIPTION
**Pull request by guoyongzhang**
follow the same idea as https://github.com/verrazzano/verrazzano/pull/673.

### CODE REVIEW

File: pkg/managedclusters/secrets.go [hunk 1]
```diff
@@ -99,3 +100,69 @@ func GetRancherCACert(kubeClientSet kubernetes.Interface) []byte {
...
 	return certSecret.Data["ca.crt"]
 }
+
+// GetRancherCredentials returns username/password from secret "verrazzano" in namespace "verrazzano-system"
+func GetRancherCredentials(kubeClientSet kubernetes.Interface) (string, string) {
+	const verrazzanoNamespace = "verrazzano-system"
+	const credentailsSecret = "verrazzano"
+	credsSecret, err := kubeClientSet.CoreV1().Secrets(verrazzanoNamespace).Get(context.TODO(), credentailsSecret, metav1.GetOptions{})
+	if err != nil {
+		zap.S().Warnf("Error getting secret %s/%s in management cluster: %s", verrazzanoNamespace, credentailsSecret, err.Error())
```
Comment (markxnelson) [inline pkg/managedclusters/secrets.go:110]: management cluster -> admin cluster
(occurs on other lines too)

File: pkg/managedclusters/secrets.go [hunk 2]
```diff
@@ -99,3 +100,69 @@ func GetRancherCACert(kubeClientSet kubernetes.Interface) []byte {
...
+	ingress, err := kubeClientSet.ExtensionsV1beta1().Ingresses(rancherNamespace).Get(context.TODO(), rancherIngressName, metav1.GetOptions{})
+	if err != nil {
+		zap.S().Warnf("Error getting ingress %s/%s in management cluster: %s", rancherNamespace, rancherIngressName, err.Error())
+		return ""
+	}
+	if ingress == nil {
+		zap.S().Warnf("Ingress %s/%s not found in management cluster", rancherNamespace, rancherIngressName)
+		return ""
+	}
+	for _, rule := range ingress.Spec.Rules {
```
Comment (markxnelson) [inline pkg/managedclusters/secrets.go:162]: won't this just take the first one every time? is this what you intended?

File: pkg/rancher/rancher.go [hunk 3]
```diff
@@ -98,3 +106,171 @@ func (c Rancher) APICall(rancherConfig Config, apiPath string, httpMethod string
...
+	})
+	if err == wait.ErrWaitTimeout {
+		if lastErr != nil {
+			err = lastErr
+		}
+	}
+	return err
+}
+
+// Retrieve poxy url from environment
```
Comment (markxnelson) [inline pkg/rancher/rancher.go:137]: typo

File: pkg/rancher/rancher.go [hunk 4]
```diff
@@ -98,3 +106,171 @@ func (c Rancher) APICall(rancherConfig Config, apiPath string, httpMethod string
...
+	}
+
+	// Add parameters
+	query := req.URL.Query()
+	for key, value := range parameterMap {
+		query.Add(key, value)
+	}
+	req.URL.RawQuery = query.Encode()
+
+	// Send the request
```
Comment (markxnelson) [inline pkg/rancher/rancher.go:240]: i wonder - do we want to use retryablehttp in here?

File: pkg/managedclusters/secrets.go (hunk 2 above)
Comment (guoyongzhang) [reply pkg/managedclusters/secrets.go:162]: yes, intended.  the first host is good.

File: pkg/managedclusters/secrets.go (hunk 1 above)
Comment (guoyongzhang) [reply pkg/managedclusters/secrets.go:110]: fixed

File: pkg/rancher/rancher.go (hunk 3 above)
Comment (guoyongzhang) [reply pkg/rancher/rancher.go:137]: fixed

File: pkg/rancher/rancher.go (hunk 4 above)
Comment (guoyongzhang) [reply pkg/rancher/rancher.go:240]: The cluster operator does this in a loop, i.e, itself is a giant retry

**Review by markxnelson — APPROVED** [review]

**Pull request merged**